### ISSUE TITLE
why Inputs requires grad?

### ISSUE THREAD

**Issue by dragen1860** [Gaoyiminggithub/Graphonomy#23]
Dear author:
  thanks for share the implementation code!
  When i dive into your code, a trivial detail confused me. as stated below:
```
        net.train()
        for ii, sample_batched in enumerate(trainloader):

            inputs, labels = sample_batched['image'], sample_batched['label']
            # Forward-Backward of the mini-batch
            inputs, labels = Variable(inputs, requires_grad=True), Variable(labels)
            global_step += inputs.data.shape[0]
```
I dnk why the inputs requires gradient? Please correct me! thank you.

**Comment by Gaoyiminggithub** [comment]
Hi, 
During training, tensors need gradient to update the weight of network.
